cpp/src/defaults.cpp

```diff
@@ -91,6 +91,25 @@ CompatMode getenv_or(std::string_view env_var_name, CompatMode default_val)
   return detail::parse_compat_mode_str(env_val);
 }
 
+template <>
+std::vector<int> getenv_or(std::string_view env_var_name, std::vector<int> default_val)
+{
+  auto* env_val = std::getenv(env_var_name.data());
+  if (env_val == nullptr) { return default_val; }
+  std::vector<int> vect;
+  std::stringstream ss(env_val);
+
+  for (int i; ss >> i;) {
+    vect.push_back(i);
+    if (ss.peek() == ',' || ss.peek() == ' ') { ss.ignore(); }
+
+    // TODO: this isn't working yet.
+    if (ss.fail()) { throw std::invalid_argument("Invalid value " + std::string(env_val)); }
+  }
+
+  return vect;
+}
+
```

Comment:
Using regex, this would work:

```suggestion
template <>
std::vector<int> getenv_or(std::string_view env_var_name, std::vector<int> default_val)
{
  auto* const env_val = std::getenv(env_var_name.data());
  if (env_val == nullptr) { return default_val; }
  std::string const int_str(env_val);
  if (int_str.empty()) { return default_val; }

  // Ensure `int_str` consists only of integers separated by commas, allowing spaces.
  std::regex const check_pattern(R"(^\s*\d+\s*(\s*,\s*\d+\s*)*$)");
  if (!std::regex_match(int_str, check_pattern)) {
    throw std::invalid_argument(std::string{env_var_name} +
                                ": invalid format, expected comma-separated integers.");
  }

  // Match every integer in `int_str`.
  std::regex const number_pattern(R"(\d+)");

  // For each match, we push_back `std::stoi(match.str())` into `ret`.
  std::vector<int> ret;
  std::transform(std::sregex_iterator(int_str.begin(), int_str.end(), number_pattern),
                 std::sregex_iterator(),
                 std::back_inserter(ret),
                 [](std::smatch const& match) -> int { return std::stoi(match.str()); });
  return ret;
}

```

Reply:
Maybe we make it even stricter and require the integers to have 3 digits. This might work (untested).
```
std::regex const check_pattern(R"(^\s*\d{3}\s*(\s*,\s*\d{3}\s*)*$)");
```

Reply:
That works perfectly, thanks!

```
KVIKIO_HTTP_STATUS_CODES="429,500,a" python -c 'import kvikio.defaults; print(kvikio.defaults.http_status_codes())'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/coder/kvikio/python/kvikio/kvikio/defaults.py", line 318, in http_status_codes
    return kvikio._lib.defaults.http_status_codes()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "defaults.pyx", line 89, in kvikio._lib.defaults.http_status_codes
ValueError: KVIKIO_HTTP_STATUS_CODES: invalid format, expected comma-separated integers.
```

I'm going to write up a quick test for this in C++ before pushing.

Reply:
Well, I haven't figured out how to run the C++ tests locally yet :/ I'll come back to the later today.

Reply:
Added in https://github.com/rapidsai/kvikio/pull/603/commits/bf33697ba450173ccd76f4de46cc2faf6308c9d2.